source/RemoteWiring/RemoteDevice.cpp

```diff
@@ -159,6 +180,7 @@ RemoteDevice::pinMode(
 	uint8_t port_mask;
 	getPinMap( pin_, &port, &port_mask );
 
+    _firmata->lock();
```

Comment:
I love this idea, I hadn't considered using the UwpFirmata object in the template.